tests/test_commit.py

```diff
@@ -326,3 +330,141 @@ def test_commit_schema_violation(cli_runner, data_working_copy):
             "nz_pa_points_topo_150k: In column 'macronated' value 'kinda' exceeds limit of 1 characters",
             "Error: Schema violation - values do not match schema",
         ]
+
+
+def test_commit_table_json_output(cli_runner, data_working_copy):
+    new_table = "test_table"
+    message = "test commit"
+    with data_working_copy("points") as (path, wc):
+        repo = KartRepo(path)
+        with repo.working_copy.tabular.session() as sess:
+            sess.execute(
+                f"""CREATE TABLE IF NOT EXISTS {new_table} (test_id int primary key, field1 text, field2 text);"""
+            )
+            sess.execute(
+                f"""INSERT INTO {new_table} (test_id, field1, field2)
+                VALUES
+                    (1, 'value1a', 'value1b'),
+                    (2, 'value2a', 'value2b'),
+                    (3, 'value3a', 'value3b'),
+                    (4, 'value4a', 'value4b'),
+                    (5, 'value5a', 'value5b');"""
+            )
+
+        r = cli_runner.invoke(["add-dataset", new_table, "-m", message, "-o", "json"])
+
+        assert r.exit_code == 0, r
+
+        author = repo.head_commit.author
+
+        expected_output: Dict[str, Dict[str, any]] = {
+        "kart.commit/v1": {
+            "commit": str(repo.head.target),
+            "abbrevCommit": str(repo.head.target)[:7],
+            # Check later: author should probably be author.name but in commit.py l284 it is author.email:
+            "author": author.email,
+            "committer": author.email,
+            "branch": "main",
+            "message": message,
+            "changes": {
+                new_table: {
+                    "meta": {
+                        "inserts": 1
+                    },
+                    "feature": {
+                        "inserts": 5
+                    }
+                }
+            },
+            "commitTime": datetime.utcnow().replace(microsecond=0).isoformat() + 'Z',
+            "commitTimeOffset": "+13:00"
```

Comment:
the utcnow() thing is going to be flaky if the test runs a bit slow, and (I think) the offset will vary depending on who runs the test.

Rather, hardcode a specific time/offset in an environment variable. There's some examples of this in the `commit` tests: https://github.com/koordinates/kart/blob/9bf9e164cf16f81ea7709a9aeafa1f68a0cf1b06/tests/test_commit.py#L292-L298